### PR TITLE
Legends handle wide maps

### LINKED ISSUE #1585: Layout: handle wide states
Split off from #1518 

## Expected Behavior
Wide states should render with a different layout: full width and horizontal legend

screen shot 2016-07-08 at 4 23 27 pm
## Actual Behavior
Wide states are confined to a narrow width and vertical legend
## Steps to reproduce the behavior
- [ ] layout changes
- [x] add 'horizontal' legend option. The difficult part here is making sure that the filtered swatches translate horizontally instead of vertically
- [ ] create list of states that are 'wide'
- [x] apply changes to 'wide' states

### PR DESCRIPTION
**Pull request by gemfarmer**
Fixes #1585 

Horizontal stacking display: [MT :sunglasses: PREVIEW](https://federalist.18f.gov/preview/18F/doi-extractives-data/wide-maps/states/TN)
Vertical side-by-side display: [UT :sunglasses: PREVIEW](https://federalist.18f.gov/preview/18F/doi-extractives-data/wide-maps/states/UT)

legends for the `<eiti-data-map>` component render a legend svg. This PR attempts to make the two column `map | legend` layout a one column layout (map and legend below it). There will also be a few other tweaks to the legend styling

Changes proposed in this pull request:
- [x] Detect if a map should have a "horizontal" or "vertical" layout
- [x] Basic horizontal implimentation 
- [x] make PRs to the [d3-legend](https://github.com/susielu/d3-legend)
  - [x] handle non-delimited quantize scale labels
  - [x] ~~enable a scale to be [filterable based its data](https://github.com/susielu/d3-legend/issues/31)~~ _instead, I added this logic to the data map component. It should eventually be removed, but works for now.
- [x] Make legend in side-by-side display more "fluffed"

/cc @ericronne 

#### Some Examples:
screen shot 2016-07-28 at 12 05 23 pm
screen shot 2016-07-28 at 12 05 14 pm
screen shot 2016-07-28 at 12 05 00 pm
screen shot 2016-07-28 at 12 04 53 pm
screen shot 2016-07-28 at 12 04 38 pm
screen shot 2016-07-28 at 12 04 25 pm
screen shot 2016-07-28 at 12 02 28 pm
screen shot 2016-07-28 at 12 02 19 pm
screen shot 2016-07-28 at 12 02 12 pm
screen shot 2016-07-28 at 12 06 49 pm

### CODE REVIEW

File: _includes/year-selector.html [hunk 1]
```diff
@@ -14,14 +14,12 @@
       {% assign __year = year | to_i %}
       {% assign s_year = _year | to_s %}
       {% if include.empty_years.size > 0 %}
-        <!-- {{ empty_years }} -->
         {% if empty_years contains s_year %}
-          <option disabled value="{{ _year }}" {% if __year == _year %}selected{% endif %}>{{ _year }} (no data)</option>
+          <option disabled value="{{ _year }}" {% if __year == _year %}selected{% endif %}>{{ _year }} no data</option>
```
Comment (gemfarmer) [inline _includes/year-selector.html]: I updated the selector text to be more inline with the mock

File: _sass/blocks/_chart-list.scss [hunk 2]
```diff
@@ -74,7 +74,7 @@
   }
 
   figcaption {
-    @include font-size(1);
+    @include font-size(0.9375);
```
Comment (gemfarmer) [inline _sass/blocks/_chart-list.scss]: * 15px

**Comment by gemfarmer** [comment]
Ready for review @ericronne, @shawnbot!

**Comment by meiqimichelle** [comment]
This is what I'm seeing for Tennessee. Looks like the legend isn't working, maybe when the default year doesn't have data? Also, the little navigation map on the side nav isn't working (maybe this work introduced a bug there, or maybe that was already known), but that's not a huge issue for now.
screenshot 2016-07-28 18 23 26

**Comment by gemfarmer** [comment]
@meiqimichelle After poking around, I think this is a Firefox issue. Neither of those appear to be bugs on Safari or Chrome. I'll look into it!

**Comment by gemfarmer** [comment]
Left to do
- [ ] left align legend text on buckets

**Comment by gemfarmer** [comment]
Mock for employment bars

pasted image at 2016_08_01 03_41 pm

**Comment by gemfarmer** [comment]
@ericronne I think all of the changes we discussed here have been addressed!

**Comment by gemfarmer** [comment]
@shawnbot ready to merge when you feel like giving it the go-ahead.

I'm not sure it would have saved me much time so far (maybe??), but I am a little nervous about dev speed for a new 💫 mobile layout if it includes the legend. 😟

**Pull request merged**